### ISSUE TITLE
Crash on calculating conversion lag adjustment

### ISSUE THREAD

**Issue by hiering** [google/app-reporting-pack#26]
Hello!
We would like to try the App Reporting Pack but we get the error `ValueError: 0 out of range`. We have tried running locally or through GCP but we always get this error. Maybe you have an idea what this is related to?

```
===fetching reports===
===core===
<jemalloc>: MADV_DONTNEED does not work (memset will be used instead)
<jemalloc>: (This is the expected behaviour if you are running under QEMU)
[2024-01-15 11:03:23][gaarf.cli.utils][INFO] Total number of customer_ids is 69, accounts=[***]
[2024-01-15 11:03:23][gaarf.cli.utils][INFO] Running queries in parallel
[2024-01-15 11:05:14][gaarf.cli.utils][INFO] /app/core/google_ads_queries/bid_budget.sql executed successfully
[2024-01-15 11:05:23][gaarf.cli.utils][INFO] /app/core/google_ads_queries/campaign_languages.sql executed successfully
[2024-01-15 11:06:30][gaarf.cli.utils][INFO] /app/core/google_ads_queries/campaign_geo_targets.sql executed successfully
[2024-01-15 11:06:30][gaarf.cli.utils][INFO] /app/core/google_ads_queries/app_campaign_settings.sql executed successfully
[2024-01-15 11:06:51][gaarf.cli.utils][INFO] /app/core/google_ads_queries/action_items.sql executed successfully
[2024-01-15 11:06:55][gaarf.cli.utils][INFO] /app/core/google_ads_queries/ocid_mapping.sql executed successfully
 [2024-01-15 11:08:22][gaarf.cli.utils][INFO] /app/core/google_ads_queries/account_campaign_ad_group_mapping.sql executed successfully
[2024-01-15 11:11:19][gaarf.cli.utils][INFO] /app/core/google_ads_queries/geo_target_constant.sql executed successfully
[2024-01-15 11:11:59][gaarf.cli.utils][INFO] /app/core/google_ads_queries/app_conversions_mapping.sql executed successfully
[2024-01-15 11:15:19][gaarf.cli.utils][INFO] /app/core/google_ads_queries/ad_group_performance.sql executed successfully
[2024-01-15 11:42:52][gaarf.cli.utils][INFO] /app/core/google_ads_queries/ad_group_conversion_split.sql executed successfully
===calculating conversion lag adjustment===
<jemalloc>: MADV_DONTNEED does not work (memset will be used instead)
<jemalloc>: (This is the expected behaviour if you are running under QEMU)
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/pandas/core/indexes/range.py", line 414, in get_loc
    return self._range.index(new_key)
ValueError: 0 is not in range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/scripts/conv_lag_adjustment.py", line 91, in <module>
    main()
  File "/app/scripts/conv_lag_adjustment.py", line 83, in main
    conv_lag_table = conv_lag_builder.calculate_reference_values()
  File "/app/scripts/src/conv_lag_builder.py", line 53, in calculate_reference_values
    cumulative_data = self.calculate_cumulative_sum_ordered_by_n(
  File "/app/scripts/src/conv_lag_builder.py", line 116, in calculate_cumulative_sum_ordered_by_n
    "daily_incremental_lag"].fillna(new_ds["pct_conv"][0])
  File "/usr/local/lib/python3.10/site-packages/pandas/core/series.py", line 1040, in __getitem__
    return self._get_value(key)
  File "/usr/local/lib/python3.10/site-packages/pandas/core/series.py", line 1156, in _get_value
    loc = self.index.get_loc(label)
  File "/usr/local/lib/python3.10/site-packages/pandas/core/indexes/range.py", line 416, in get_loc
    raise KeyError(key) from err
KeyError: 0
```

**Comment by drLoom** [comment]
This behaviour occured when all rows of df_group have lag_number = Nan, so group by will result in []
`
def calculate_reference_values(self) -> pd.DataFrame:
...
conversion_lag_table = []
        for _, df_group in grouped:
            grouped_data = self.calculate_conversions_by_name_network_lag(
                df_group)
`

[PR-28](https://github.com/google/app-reporting-pack/pull/28) But, probably, better solution is to filter data before

**Comment by AndreyMarkinPPC** [comment]
Fixed, could refer to version 1.4.0 - https://github.com/google/app-reporting-pack/tree/1.4.0
